### PR TITLE
Use instantiate

### PR DESCRIPTION
**Pull request by matthuhiggins**
Use klass.instantiate to load persisted records. Use klass.new results in `new_record? == true`, `persisted? == false` and ActiveModel::Dirty attributes.

### CODE REVIEW

File: lib/elastic_record/search_hits.rb [hunk 1]
```diff
@@ -19,7 +19,9 @@ def to_ids
 
     def to_records
       if model.elastic_index.load_from_source
-        hits.map { |hit| load_from_hit(hit) }
+        hits.map do |hit|
+          model.respond_to?(:instantiate) ? model.instantiate(hit) : load_from_hit(hit)
+        end
```
Comment (bullfight) [inline lib/elastic_record/search_hits.rb]: Here the `hit` is in the form:

```ruby
Person.instantiate('_id' => 1234, '_source' => { 'first_name' => "foo" })
```

[`Superstore::Persistence`](https://github.com/data-axle/superstore/blob/master/lib/superstore/persistence.rb) expects the hash to be in the form:

```ruby
Person.instantiate('id' => 1234, 'document' => { 'first_name' => "foo" })
```

https://github.com/data-axle/superstore/blob/6208cb519ebb5de44fdc3db4ad14ea7e8b5cc977/lib/superstore/persistence.rb#L22-L27

Comment (kunruh9) [reply lib/elastic_record/search_hits.rb]: I just noticed this as well. But this change can't be for our specific needs and knows nothing about superstore.

Comment (kunruh9) [reply lib/elastic_record/search_hits.rb]: Looking into leveraging this change elsewhere.

Comment (matthuhiggins) [reply lib/elastic_record/search_hits.rb]: Right, let's fix instantiate then

**Comment by matthuhiggins** [comment]
@bullfight See https://github.com/data-axle/superstore/pull/39

File: lib/elastic_record/search_hits.rb [hunk 2]
```diff
@@ -19,7 +19,9 @@ def to_ids
 
     def to_records
       if model.elastic_index.load_from_source
-        hits.map { |hit| load_from_hit(hit) }
+        hits.map do |hit|
+          model.respond_to?(:instantiate) ? model.instantiate(hit['_source'].update('id' => hit['_id'])) : load_from_hit(hit)
+        end
```
Comment (bullfight) [inline lib/elastic_record/search_hits.rb]: <bikeshed> `instantiate_from_hit(hit)` </bikeshed>

**Comment by matthuhiggins** [comment]
I like bike shedding

**Pull request merged**